home/tests/tests.py

```diff
@@ -55,3 +58,56 @@ def test_do_not_change_queryset_when_parent_page_is_not_a_section(self):
         pages_after = limit_page_chooser(pages_before, request)
 
         self.assertEqual(pages_after, pages_before)
+
+
+class MediaTranslationTest(TestCase):
+    def setUp(self):
+        self.site = Site.objects.get(is_default_site=True)
+        self.en_home_page = self.site.root_page
+        self.en_article = ArticleFactory(
+            parent=self.en_home_page,
+            body__0__media=MediaFactory(type='video'),
+            body__1__media=MediaFactory(type='audio'),
+        )
+
+    def test_media_block_translation_of_english_language(self):
+        response = self.client.get(self.en_article.url)
+        soup = BeautifulSoup(response.content)
+        video_text = soup.find("p", {"class": "article__content--video"}).text.strip()
+        video_link = soup.find("p", {"class": "article__content--video"}).find("a").get("href")
+        audio_text = soup.find("p", {"class": "article__content--audio"}).text.strip()
+        audio_link = soup.find("p", {"class": "article__content--audio"}).find("a").get("href")
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(video_text, 'If you cannot view the above video, you can instead download it.')
+        self.assertEqual(video_link, self.en_article.body[0].value.url)
+        self.assertEqual(audio_text, 'If you cannot listen to the above audio, you can instead download it.')
+        self.assertEqual(audio_link, self.en_article.body[1].value.url)
+
+    def test_media_block_translation_of_bengali_language(self):
+        bn_locale = LocaleFactory(language_code='bn')
+        bn_translation_creator = TranslationCreator(user=None, target_locales=[bn_locale])
+        bn_translation_creator.create_translations(self.en_article)
+        bn_article = self.en_article.get_translation(bn_locale)
+
+        TranslationEntry.objects.create(
+            original="If you cannot view the above video, you can instead %(start_link)sdownload it%(end_link)s.",
+            translation="উপরের ভিডিও দেখা না গেলে %(start_link)s এর পরিবর্তে এটা %(end_link)s ডাউনলোড করুন",
+            language=bn_locale.language_code)
+        TranslationEntry.objects.create(
+            original="If you cannot listen to the above audio, you can instead %(start_link)sdownload it%(end_link)s.",
+            translation="উপরের অডিও শুনতে না পেলে %(start_link)s এর পরিবর্তে এটা %(end_link)s ডাউনলোড করুন",
+            language=bn_locale.language_code)
+
+        response = self.client.get(bn_article.url)
+        soup = BeautifulSoup(response.content)
+        video_text = soup.find("p", {"class": "article__content--video"}).text.strip()
+        video_link = soup.find("p", {"class": "article__content--video"}).find("a").get("href")
+        audio_text = soup.find("p", {"class": "article__content--audio"}).text.strip()
+        audio_link = soup.find("p", {"class": "article__content--audio"}).find("a").get("href")
```

Comment:
There is another approach to writing integration/system tests for Django templates, it checks if correct values are available in the context (if necessary) and the response contains what is expected irrespective of the location (no need to drill down to exact HTML element). The issue with this approach is that it isn't as robust as the current approach. Both have their own pros and cons but let me know if we should move to this approach it will simplify our tests.
For reference:
- https://test-driven-django-development.readthedocs.io/en/latest/03-views.html
- https://stackoverflow.com/questions/10608681/how-to-test-render-to-template-functions-in-django-tdd
- https://github.com/wagtail/wagtail/blob/main/wagtail/users/tests/test_admin_views.py#L109

Reply:
If it makes sense to use a different approach, then I'm happy for you to take it. I also don't think there is a one-size-fits-all solution. My problem was with how this approach had been implemented i.e. with lots of duplication.